jobmon_client/src/jobmon/client/swarm/workflow_run.py

```diff
@@ -515,12 +515,23 @@ def run(
                 # If the workflow run status was updated asynchronously, terminate
                 # all active task instances and error out.
                 if self.status in terminating_states:
-                    logger.warning(
-                        f"Workflow Run set to {self.status}. Attempting graceful shutdown."
+                    wait_states = [
```

Comment:
Question along the same vein as I left in fsm/array.py. Are we certain about these wait states? Do we need to add any more e.g. Registered.